assets/javascripts/kitten/components/navigation/header-nav/index.js

```diff
@@ -0,0 +1,165 @@
+import { Nav } from './components/nav'
+import { UserMenu } from './components/user-menu'
+import { Context } from './components/context'
+import { Right } from './components/right'
+import { Centered } from './components/centered'
+import { Unlogged } from './components/unlogged'
+import { Logged } from './components/logged'
+import { Hidden } from './components/hidden'
+import { QuickAccessLink } from './components/quick-access-link'
+import {
+  MOBILE_HEADER_HEIGHT,
+  TABLET_HEADER_HEIGHT,
+  DESKTOP_HEADER_HEIGHT,
+} from './config'
+import { StickyContainer } from '../../../components/grid/sticky-container'
+import COLORS from '../../../constants/colors-config'
+
+const StyledStickyContainer = styled(StickyContainer)`
+  ${({ isMenuExpanded }) =>
+    isMenuExpanded &&
+    css`
+      transition: none;
+    `}
+
+  .k-Spacer + & {
+    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
+  }
+`
+
+const Container = styled.div`
+  position: relative;
+  z-index: ${({ isMenuExpanded, zIndex }) =>
+    isMenuExpanded ? zIndex.headerWithOpenMenu : zIndex.header};
+`
+
+const Header = styled.header`
+  width: 100vw;
+  overflow: hidden;
+  box-sizing: border-box;
+
+  &,
+  .quickAccessLink {
+    height: ${MOBILE_HEADER_HEIGHT};
+    background: ${({ updateBackground }) =>
+      updateBackground ? COLORS.background3 : COLORS.background1};
+
+    @media (min-width: ${ScreenConfig.S.min}px) {
+      height: ${TABLET_HEADER_HEIGHT};
+    }
+
+    @media (min-width: ${ScreenConfig.L.min}px) {
+      height: ${DESKTOP_HEADER_HEIGHT};
+    }
+  }
+`
+
+const HeaderNav = ({
+  children,
+  id,
+  isFixed,
+  isLogged,
+  quickAccessProps,
+  zIndexConfig,
+}) => {
+  const [isLoggedState, setIsLogged] = useState(isLogged)
+  const [idState, setId] = useState(id)
+  const [isMenuExpanded, setMenuExpanded] = useState(false)
+  const [menuExpandBy, setMenuExpandBy] = useState(null)
+  const stickyContainerRef = useRef(null)
+
+  useEffect(() => {
+    setIsLogged(isLogged)
+  }, [isLogged])
+
+  useEffect(() => {
+    setId(idState)
+  }, [id])
+
+  const updateHeaderBackground = () => /UserMenu/.test(menuExpandBy)
+
+  const callOnToggle = ({ isExpanded, expandBy }) => {
+    if (!isExpanded) stickyContainerRef.current.setSticky()
+
+    setMenuExpanded(isExpanded)
+    setMenuExpandBy(expandBy)
+  }
+
+  return (
+    <Context.Provider
+      value={{
+        isLogged: isLoggedState,
+        id: idState,
+        expandBy: menuExpandBy,
+        callOnToggle,
+      }}
+    >
+      <Container zIndex={zIndexConfig}>
+        <StyledStickyContainer
+          ref={stickyContainerRef}
+          isSticky={isFixed || isMenuExpanded ? 'always' : 'topOnScrollUp'}
+          isMenuExpanded={isMenuExpanded}
+        >
+          <Header
+            role="banner"
+            id={idState}
+            className="k-HeaderNav"
+            updateBackground={updateHeaderBackground()}
+          >
+            <QuickAccessLink
```

Comment:
Est-ce qu'il y a une raison pour laquelle `QuickAccessLink` est dans `Header` et pas plus haut dans le DOM ? Je ne sais pas très bien ce qui est préconisé, je me demande juste s'il y a une raison que je ne connais pas :)

Reply:
Dans tous les headers que j'ai pu regarder, le `QuickAccessLink` est toujours dans le header, que ce soit sur [adrianroselli.com](https://adrianroselli.com/2017/10/dont-use-aria-menu-roles-for-site-nav.html#Skip) ou même [Github](https://github.com/). Ce qui me semble logique d'ailleurs si on veut que cet élément soit bien positionné lorsque la navigation est sticky. C'est aussi là où on l'avait mis d'origine sur le header qui existe actuellement sur KissKiss.